src/DependencyInjection/SetonoSyliusOrderEditExtension.php

```diff
@@ -25,6 +28,20 @@ public function load(array $configs, ContainerBuilder $container): void
 
     public function prepend(ContainerBuilder $container): void
     {
+        $container->prependExtensionConfig('winzou_state_machine', [
```

Comment:
I am much bigger fan of doing this with events and event subscribers. Let me hear your thoughts

Reply:
Well, there is one important reason to handle it with a state machine callback rather than the e.g. `pre_complete` event. The events are thrown only when the order is completed with the ResourceBundle. This means they could not be there if it's completed by API or by some custom action. So relying on the state machine we actually rely on the lower level of abstraction, that have a bigger coverage of cases.
We could in theory rely on the doctrine events, but I don't like handling of them and not being able to be precise what exactly we're listening for (only for completion, not _any_ order update)

🖖

Reply:
I meant the events thrown in the resource controller, e.g. `sylius.order.pre_complete`

Reply:
Hence the first part of my comment. If we subscribe only to those events, the functionality would not work if the order is completed e.g. via the API - which is probably not something we want

Reply:
Aaah, yeah, okay. Agreed

Reply:
I am pretty sure the state machine also fires events. Could we subscribe to that instead?

It's just I am big non-fan of configuration if I can have code that does the same thing, because my experience is you end up with fewer bugs because tools will help you refactor, autocomplete, find type errors, etc.

Reply:
State machine events - good one, never used them 😅 What would you say, to merge the current implementation as it is, and switch to state machine events in a separate PR? Just to not make this PR even bigger 🖖